### PR TITLE
Add LongForm dataset (better than Alpaca)

### PR DESCRIPTION
**Pull request by rasbt**
Adds the LongForm dataset.

LongForm is a semi-synthetic dataset based on raw text corpora for which the instructions were generated via an LLM. For more details about the instruction-generation process, please refer to the [LongForm research paper](https://arxiv.org/abs/2304.08460) by Köksal et al. According to the research paper, a Llama 7B model trained on LongForm achieves substantially better performance than the same Llama model trained on the 2x larger Alpaca dataset.

LongForm consists of 23,652 training samples, 2,042 validation samples, and 2,045 test samples. (In Lit-GPT, the validation samples are currently not used.)

The more detailed dataset composition is as follows based on a table taken from the [dataset repository](https://github.com/akoksal/LongForm):

| **Type**               | **Source**     | **Number of Examples** |
|------------------------|----------------|------------------------|
| **Corpora**            | C4             | 10,000                 |
|                        | Wikipedia      | 5,000                  |
| **Structured Corpora** | Stack Exchange | 4,380                  |
|                        | WikiHow        | 2,500                  |
| **Tasks**              | NIv2           | 3,684                  |
|                        | Big Bench      | 600                    |
|                        | BEA-GEC        | 1,203                  |
|                        | Enron          | 372                    |
| **Total**              |                | 27,739                 |
|  |   |  |
| **Train**              |                | 23,652                 |
| **Validation**         |                | 2,042                  |
| **Test**               |                | 2,045                  |

License information is not provided but would depend on the individual subsets listed above.

### CODE REVIEW

**Comment by rasbt** [comment]
@carmocca There seem to be some CI issues unrelated to the PR. Perhaps issues elsewhere too since they seem to be missing-authtoken related.

File: tutorials/prepare_dataset.md [hunk 1]
```diff
@@ -112,100 +142,7 @@ Please read the [tutorials/finetune_*.md](../tutorials) documents for more infor
 
 > [!IMPORTANT]
 > By default, the maximum sequence length is obtained from the model configuration file. In case you run into out-of-memory errors, especially in the cases of LIMA and Dolly,
-> you can try to lower the context length by preparing the dataset with a fixed max length, for example, `python scripts/prepare_lima.py --max_seq_length 2048`. Alternatvively, you can edit the  [`finetune/lora.py` file](https://github.com/Lightning-AI/lit-gpt/blob/main/finetune/lora.py#L37) and change `override_max_seq_length = None` to `override_max_seq_length = 2048`.
-
-&nbsp;
-
-## Preparing Custom Datasets for Instruction Finetuning
```
Comment (carmocca) [inline tutorials/prepare_dataset.md:119]: Is all of this removed on purpose?

**Review by carmocca — COMMENTED** [review]
I had to renew the token. Should be fixed now

File: tutorials/prepare_dataset.md (hunk 1 above)
Comment (rasbt) [reply tutorials/prepare_dataset.md:119]: omg sry, definitely not intentional. Good that you caught this. Will add back momentarily

**Comment by rasbt** [comment]
Ok fixed

File: tutorials/prepare_dataset.md [hunk 2]
```diff
@@ -73,6 +74,36 @@ python scripts/prepare_dolly.py \
...
+| **Corpora**            | C4             | 10,000                 |
+|                        | Wikipedia      | 5,000                  |
+| **Structured Corpora** | Stack Exchange | 4,380                  |
+|                        | WikiHow        | 2,500                  |
+| **Tasks**              | NIv2           | 3,684                  |
+|                        | Big Bench      | 600                    |
+|                        | BEA-GEC        | 1,203                  |
+|                        | Enron          | 372                    |
+| **Total**              |                | 27,739                 |
+|  |   |  |
```
Comment (carmocca) [inline tutorials/prepare_dataset.md:99]: This is intended right? As a separator

**Review by carmocca — APPROVED** [review]
LGTM

**Pull request merged**